File: WebApplication2/WebApplication1/Modules/TreesModule.cs
using Nancy.ModelBinding;
using WebApplication1.Services;
using WebApplication1.Model;
using Nancy;

namespace WebApplication1.Modules
{
    public class TreesModule : NancyModule
    {

        public TreesModule(TreeService treeService)
        {
            Get["/trees"] = parameters => View[new TreesModel() {Trees = treeService.AllTrees()}];
            Get["/trees/{Id}"] = parameters =>
                {
                    int result;
                    var isInteger = int.TryParse(parameters.id, out result);
                    var tree = treeService.FindById(result);
                    if (isInteger && tree != null)
                    {
                        return View[tree];
                    }
                    else
                    {
                        return HttpStatusCode.NotFound;
                    }
                };
            Get["/trees/add/"] = parameters => View["AddTree.cshtml", new TreeModel()];
            Post["/trees/add/"] = parameters =>
            {
                var tree = this.Bind<TreeModel>();
                treeService.Add(tree);
                return Response.AsRedirect("/trees");
            };
        }
    }
}

File: WebApplication2/WebApplication2/Modules/TreesModule.vb
Imports WebApplication2.Services
Imports WebApplication2.Model
Imports Nancy

Namespace Modules

    Public Class TreesModule
        Inherits NancyModule

        Public Sub New(ByVal treeService As TreeService)
            MyBase.Get("/trees") = Function(parameters)
                                       Return View(New TreesModel() With {.Trees = treeService.AllTrees()})

                                   End Function
            MyBase.Get("/trees/{Id}") = Function(parameters)
                                            Dim result As Integer
                                            Dim isInteger = Integer.TryParse(parameters.id, result)
                                            Dim tree = treeService.FindById(result)
                                            If isInteger AndAlso tree IsNot Nothing Then
                                                Return View(tree)
                                            Else
                                                Return HttpStatusCode.NotFound
                                            End If
                                        End Function

        End Sub

    End Class
End Namespace